src/v/cluster/cluster_discovery.cc

```diff
@@ -148,18 +162,185 @@ ss::future<bool> cluster_discovery::dispatch_node_uuid_registration_to_seeds(
     co_return false;
 }
 
-/*static*/ std::optional<node_id>
+ss::future<cluster_bootstrap_info_reply>
+cluster_discovery::request_cluster_bootstrap_info_single(
+  const net::unresolved_address addr) const {
+    vlog(clusterlog.info, "Requesting cluster bootstrap info from {}", addr);
+    _as.check();
+    cluster_bootstrap_info_reply reply;
+    for (auto repeat_jitter = simple_time_jitter<model::timeout_clock>(1s);;) {
+        result<cluster_bootstrap_info_reply> reply_result(
+          std::errc::connection_refused);
+        try {
+            reply_result = co_await do_with_client_one_shot<
+              cluster_bootstrap_client_protocol>(
+              addr,
+              config::node().rpc_server_tls(),
+              2s,
+              [](cluster_bootstrap_client_protocol c) {
+                  return c
+                    .cluster_bootstrap_info(
+                      cluster_bootstrap_info_request{},
+                      rpc::client_opts(rpc::clock_type::now() + 2s))
+                    .then(&rpc::get_ctx_data<cluster_bootstrap_info_reply>);
+              });
+            if (reply_result) {
+                reply = std::move(reply_result.value());
```

Comment:
can we `co_return reply.value()` here ? We wouldn't have to mutate reply local variable.